### PR TITLE
Restored support for the https_port clientca option

### PR DESCRIPTION
**Pull request by rousskov**
Probably missing since commit 621f429.

### CODE REVIEW

**Comment by rousskov** [comment]
Disclaimer: This PR has not been tested at all and may need some GnuTLS-specific adjustments. I am just trying to help [folks](http://lists.squid-cache.org/pipermail/squid-users/2018-July/018658.html) who suffer from this bug.

@yadij, if this fix needs more work, please take over the branch.

**Comment by yadij** [comment]
I think the call to loadClientCaFile() needs to be done before the one to updateContextConfig().

PS. How exactly would I take over a measurement-factory branch? is there something more particular than creating a different PR from my fork?

**Comment by rousskov** [comment]
> How exactly would I take over a measurement-factory branch

Just commit to the PR branch as you see fit. If you prefer, feel free to close this PR, of course.

**Comment by yadij** [comment]
"error: pathspec 'measurement-factory/missing-clientca-support' did not match any file(s) known to git"

**Comment by yadij** [comment]
Seems that the github online editor was sufficient though.

**Comment by rousskov** [comment]
I do not know what led to that git error, but it sounds like you did not specify `git push --repo` while also not telling your git to recognize `measurement-factory` as a remote git repository (`git remote add`). The exact sequence of commands to push to a remote repository branch depends on how you setup your work environment. In some cases, no special options/commands are needed at all. The next time I need to do that, I will try to remember to update GitHints with an example. Glad you found a workaround.

File: src/security/ServerOptions.cc [hunk 1]
```diff
@@ -264,13 +266,13 @@ Security::ServerOptions::createStaticServerContext(AnyP::PortCfg &port)
         }
 #endif
 
+        if (!loadClientCaFile())
```
Comment (rousskov) [inline src/security/ServerOptions.cc]: If this call must go before updateContextConfig() for some reason, we should at least add a corresponding comment here.

Comment (yadij) [reply src/security/ServerOptions.cc]: done

**Comment by yadij** [comment]
Marking this for merge since it resolves the clear issues loading the CA in config.

AFAIK there may still be issues around triggering the client cert request - one of two users replicating that failure found their issue was causd by unrelated config setting, the other has not responded yet. Further investigation there requires these changes first.

**Pull request closed without merging**